### PR TITLE
lemon 1.3.1 :snowflake:

### PR DESCRIPTION
**Pull request by lorepirri**
lemon 1.3.1 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-5143]
- https://lemon.cs.elte.hu/trac/lemon/wiki/Development
- conda-forge: https://github.com/conda-forge/lemon-feedstock/blob/main/recipe/meta.yaml

### Explanation of changes:

- use the official tarball ~can't access it very well~ added to allowed domains
- use patches from conda-forge
- add patch to allow C++17 and C++20
- switch to `ninja` for `win`
- move old patches in `patches` folder

### Dependency for

- AnacondaRecipes/cspy-feedstock#1

[PKG-5143]: https://anaconda.atlassian.net/browse/PKG-5143?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -2,57 +2,61 @@
...
-    - coincbc       # [unix]
-    - graphviz
-    - bzip2
-    - ghostscript   # [unix]
-  run:
-    - glpk
-    - coincbc       # [unix]
+    - glpk           # [not (win or s390x)]
+    - coincbc        # [unix]
+    - coin-or-utils  # [unix]
```
Comment (Jrice1317) [inline recipe/meta.yaml:59]: Where does lp and mip come in like clp and cbc? [Lines 99-107](https://lemon.cs.elte.hu/hg/lemon-1.3/file/e5af35e6c93f/CMakeLists.txt#l102)

File: recipe/meta.yaml [hunk 2]
```diff
@@ -2,57 +2,61 @@
 {% set version = "1.3.1" %}
 
 package:
-  name: {{ name|lower }}
+  name: {{ name }}
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
   url: https://github.com/k-dominik/lemon/archive/{{ version }}.tar.gz
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: It's quite suspicious to rely on this _unofficial_ clone of the lemon graph library. I can easily download tarballs from the branched variant of the archive https://lemon.cs.elte.hu/hg/lemon-1.3/archive/bdabbf66b2ad.tar.bz2. Could you replace it? Just leave a note that we need to manually check the source url

Comment (lorepirri) [reply recipe/meta.yaml]: @skupr-anaconda we get the same error as I was getting using the source repository link, after a few downloads:

```
https://lemon.cs.elte.hu/pub/sources/{{ name }}-{{ version }}.tar.gz
```

with: `https://lemon.cs.elte.hu/hg/lemon-1.3/archive/bdabbf66b2ad.tar.bz2`

error:

```
07/10/24 16:13:30  INFO   WARNING: Error: HTTP 000 CONNECTION FAILED for url <https://lemon.cs.elte.hu/hg/lemon-1.3/archive/bdabbf66b2ad.tar.bz2>                                                            
                   INFO   Elapsed: -                                                                                                                                                                         
                   INFO                                                                                                                                                                                      
                   INFO   An HTTP error occurred when trying to retrieve this URL.                                                                                                                           
                   INFO   HTTP errors are often intermittent, and a simple retry will get you on your way.                                                                                                   
```

Do we have a way to cache it somewhere or create our own archive in these cases?

File: recipe/meta.yaml (hunk 1 above)
Comment (lorepirri) [reply recipe/meta.yaml:59]: They should come from `coin` itself like when using `glpk` in https://lemon.cs.elte.hu/hg/lemon-1.3/file/e5af35e6c93f/CMakeLists.txt#l85

I am not 100% sure of this. The build process was complaining of the missing pieces that I've added. The rest was found.

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Hmm, not good

Comment (lorepirri) [reply recipe/meta.yaml]: I posted something on slack, let's move it there.

Comment (lorepirri) [reply recipe/meta.yaml]: problem solved

File: recipe/meta.yaml [hunk 3]
```diff
@@ -2,57 +2,66 @@
...
 requirements:
   build:
-    - cmake
-    - make
-    - doxygen
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
+    - cmake
+    - ninja
+    - make           # [not win]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: We can use ninja on linux and mac too.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -2,57 +2,66 @@
...
   host:
-    - glpk
-    - coincbc       # [unix]
-    - graphviz
-    - bzip2
-    - ghostscript   # [unix]
-  run:
-    - glpk
-    - coincbc       # [unix]
+    - glpk           # [not (win or s390x)]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
    - glpk           # [not s390x]
```
`glpk` is available on Windows.

**Review by JeanChristopheMorinPerso — APPROVED** [review]
Great work on this one!

File: recipe/meta.yaml (hunk 3 above)
Comment (lorepirri) [reply recipe/meta.yaml]: done, thanks!

**Review by Jrice1317 — APPROVED** [review]

**Pull request merged**